SpatialGDK/Source/SpatialGDK/Private/EngineClasses/Components/ActorInterestQueryComponent.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/Components/ActorInterestQueryComponent.h"
+
+#include "Schema/Interest.h"
+#include "Interop/SpatialClassInfoManager.h"
+
+UActorInterestQueryComponent::UActorInterestQueryComponent()
+{
+	PrimaryComponentTick.bCanEverTick = false;
+}
+
+void UActorInterestQueryComponent::CreateQueries(const USpatialClassInfoManager& ClassInfoManager, const SpatialGDK::QueryConstraint& AdditionalConstraints, TArray<SpatialGDK::Query>& OutQueries) const
+{
+	for (const auto& QueryData : Queries)
+	{
+		if (!QueryData.Constraint)
+		{
+			continue;
+		}
+
+		SpatialGDK::Query NewQuery{};
+		if (AdditionalConstraints.IsValid())
+		{
+			SpatialGDK::QueryConstraint ComponentConstraints;
+			QueryData.Constraint->CreateConstraint(ClassInfoManager, ComponentConstraints);
+		
+			NewQuery.Constraint.AndConstraint.Add(ComponentConstraints);
+			NewQuery.Constraint.AndConstraint.Add(AdditionalConstraints);
+		}
+		else
+		{
+			QueryData.Constraint->CreateConstraint(ClassInfoManager, NewQuery.Constraint);
+		}
+		NewQuery.Frequency = QueryData.Frequency;
+
+		if (NewQuery.Constraint.IsValid())
+		{
+			OutQueries.Push(NewQuery);
```

Comment:
Could we cache this since we don't support changing this at runtime?

Reply:
👍 

(might need to do as cleanup -- in meetings all day today. will either fix or make a jira today)

Reply:
Actually, I'm not convinced that caching is worthwhile here - this call should only happen when an actor is created, right? (via InterestFactory::CreateInterest)

Reply:
https://improbableio.atlassian.net/browse/UNR-1649

Reply:
If the Actor's owner changes, won't be create this again?